src/services/codefixes/importFixes.ts

```diff
@@ -678,11 +678,16 @@ namespace ts.codefix {
         if (a.kind !== ImportFixKind.UseNamespace && b.kind !== ImportFixKind.UseNamespace) {
             return compareBooleans(allowsImportingSpecifier(b.moduleSpecifier), allowsImportingSpecifier(a.moduleSpecifier))
                 || compareNodeCoreModuleSpecifiers(a.moduleSpecifier, b.moduleSpecifier, importingFile, program)
+                || compareBooleans(isOnlyDotsAndSlashes(a.moduleSpecifier), isOnlyDotsAndSlashes(b.moduleSpecifier))
                 || compareNumberOfDirectorySeparators(a.moduleSpecifier, b.moduleSpecifier);
         }
         return Comparison.EqualTo;
     }
 
+    function isOnlyDotsAndSlashes(path: string) {
+        return !/[^.\/]/g.test(path);

```

Comment:
Just avoid the regex

```suggestion
        for (let i = 0; i < path.length; i++) {
            const ch = path.charCodeAt(i);
            if (i !== CharacterCodes.dot && i !== CharacterCodes.slash) return false;
        }
        return true;
```

Reply:
Set up a test on https://jsbench.me (can't share without giving them my email) and that's 25% slower.

Reply:
Fair.

Reply:
If you do care about perf though, consider hoisting the regex out though.

Reply:
I tested that too, and it was only 5% faster 😉 I’ll probably do it anyway.

Reply:
![SanicTheHedgehobGIF](https://user-images.githubusercontent.com/972891/149441025-b98c602c-0afa-476b-96fb-37893032eaf6.gif)